### PR TITLE
Remove some goog

### PR DESCRIPTION
**Pull request by ger-benjamin**
For: #2180

Replaced: 
 - goog.object (foreach, clone, extend)
 - goog.dom

Still existing: 
 - goog.Math
 - goog.fx,
 - goog.uri
 - goog.event
 - goog.length (special)

And (goog.asserts, goog.provide, goog.requires)

### CODE REVIEW

File: contribs/gmf/src/directives/mobilemeasurepoint.js [hunk 1]
```diff
@@ -293,11 +294,11 @@ gmf.MobileMeasurePointController.prototype.getMeasure_ = function() {
...
         ctn.appendChild(childEl);
       }
-    }, this);
+    }
 
-    const previousCtn = goog.dom.getElementByClass(className, el);
-    if (previousCtn) {
-      previousCtn.remove();
+    const previousCtn = el.getElementsByClassName(className);
+    if (previousCtn.length > 0) {
```
Comment (gberaudo) [inline contribs/gmf/src/directives/mobilemeasurepoint.js]: The test must be `if (previousCtn.length > 0 && previousCtn[0])`.
I am also fine with the shorter `if (previousCtn.length && previousCtn[0])`.

File: contribs/gmf/src/directives/profile.js [hunk 2]
```diff
@@ -305,7 +305,7 @@ gmf.ProfileController = function($scope, $http, $element, $filter,
   if (optionsFn) {
     const options = optionsFn();
     goog.asserts.assertObject(options);
-    goog.object.extend(this.profileOptions, options);
+    ol.obj.assign(this.profileOptions, options);
```
Comment (gberaudo) [inline contribs/gmf/src/directives/profile.js]: Must require `ol.obj`.

File: contribs/gmf/src/directives/search.js [hunk 3]
```diff
@@ -304,7 +304,7 @@ gmf.SearchController = function($scope, $compile, $timeout, gettextCatalog,
     highlight: true
   });
 
-  goog.object.extend(this.options, this.scope_['typeaheadOptions'] || {});
+  ol.obj.assign(this.options, this.scope_['typeaheadOptions'] || {});
```
Comment (gberaudo) [inline contribs/gmf/src/directives/search.js]: Idem must require `ol.obj`.

File: src/ol-ext/format/featurehash.js [hunk 4]
```diff
@@ -1135,25 +1135,21 @@ ngeo.format.FeatureHash.prototype.writeFeatureText = function(feature, opt_optio
...
-          const encoded = encodeURIComponent(
-              `${key.replace(/[()'*]/g, '_')}*${
-              value.toString().replace(/[()'*]/g, '_')}`);
-          encodedProperties.push(encoded);
-        }
-      }));
+  const propFunction = this.propertiesFunction_(feature);
+  for (const key in propFunction) {
+    const value = propFunction[key];
+    if (value !== undefined &&
```
Comment (gberaudo) [inline src/ol-ext/format/featurehash.js]: You might want to use a single line for the test condition.

File: src/ol-ext/interaction/measurearea.js [hunk 5]
```diff
@@ -29,11 +29,16 @@ ngeo.interaction.MeasureArea = function(format, opt_options) {
...
-      goog.dom.createDom('SPAN', {},
-          'Click to continue drawing the polygon.',
-          goog.dom.createDom('BR'),
-          'Double-click or click starting point to finish.');
+  this.continueMsg;
+  if (options.continueMsg !== undefined) {
+    this.continueMsg = options.continueMsg;
+  } else {
+    this.continueMsg = document.createElement('span');
+    this.continueMsg.textContent = 'Click to continue drawing the polygon.';
```
Comment (gberaudo) [inline src/ol-ext/interaction/measurearea.js]: Strange, I am wondering how this gets translated.

File: src/ol-ext/interaction/measurearea.js [hunk 6]
```diff
@@ -29,11 +29,16 @@ ngeo.interaction.MeasureArea = function(format, opt_options) {
...
+    const br = document.createElement('br');
+    br.textContent = 'Double-click or click starting point to finish.';
+    this.continueMsg.append(br);
```
Comment (gberaudo) [inline src/ol-ext/interaction/measurearea.js]: Should be `appendChild`

File: src/ol-ext/interaction/measurelength.js [hunk 7]
```diff
@@ -29,11 +29,16 @@ ngeo.interaction.MeasureLength = function(format, opt_options) {
...
-          'Double-click or click last point to finish.');
+  this.continueMsg;
+  if (options.continueMsg !== undefined) {
+    this.continueMsg = options.continueMsg;
+  } else {
+    this.continueMsg = document.createElement('span');
+    this.continueMsg.textContent = 'Click to continue drawing the line.';
+    const br = document.createElement('br');
+    br.textContent = 'Double-click or click last point to finish.';
+    this.continueMsg.append(br);
```
Comment (gberaudo) [inline src/ol-ext/interaction/measurelength.js]: Should be `appendChild`.

File: contribs/gmf/src/directives/mobilemeasurepoint.js (hunk 1 above)
Comment (ger-benjamin) [inline contribs/gmf/src/directives/mobilemeasurepoint.js]: getElementsByClassName return always an array (but empty if nothing match)

Comment (ger-benjamin) [inline contribs/gmf/src/directives/mobilemeasurepoint.js]: And `previousCtn[0]` return the first element or undefined.

File: src/ol-ext/interaction/measurearea.js (hunk 5 above)
Comment (ger-benjamin) [inline src/ol-ext/interaction/measurearea.js]: That's right... I'll take a look

File: contribs/gmf/src/directives/mobilemeasurepoint.js (hunk 1 above)
Comment (ger-benjamin) [inline contribs/gmf/src/directives/mobilemeasurepoint.js]: getElementsByClassName return always an array. So `if (previousCtn[0]) {}` is enough.

File: contribs/gmf/src/directives/profile.js (hunk 2 above)
Comment (ger-benjamin) [inline contribs/gmf/src/directives/profile.js]: Done

File: contribs/gmf/src/directives/search.js (hunk 3 above)
Comment (ger-benjamin) [inline contribs/gmf/src/directives/search.js]: Done

File: src/ol-ext/interaction/measurearea.js (hunk 5 above)
Comment (ger-benjamin) [inline src/ol-ext/interaction/measurearea.js]: No magic there. That's not translated :upside_down_face:

**Pull request merged**